### PR TITLE
Feature/integrate prod

### PR DESCRIPTION
**Pull request by zherbz**
Migration from dummy/static datasets to stac-prod datasets hosted on s3.

**Updated datasets now pointing to s3 through DuckDB:**
- Basins
- Reference Lines
- Reference Points
- Boundary Condition Lines

**Datasets that still need to be migrated:**
- Gages (read from stac API)
- Dams (read from stac API)
- COGs (read from s3 but only test data)

**New App Features:**
- Added map legend
- Restructured map popovers using custom container styling
- Added COGs back to map
- Added BC Lines as new map layer
- Added model thumbnail images when selecting a basin

### CODE REVIEW

**Review by thwllms — CHANGES_REQUESTED** [review]
Nice improvements!

File: app/src/pages/single_event.py [hunk 1]
```diff
@@ -401,39 +527,452 @@ def single_event():
...
+                        st.session_state["s3_conn"],
+                        st.session_state["pilot"],
+                        feature_gage_id,
+                        st.session_state["calibration_event"],
+                    )
+                    obs_flow_ts.rename(columns={"flow": "obs_flow"}, inplace=True)
+                    gage_flow_ts = obs_flow_ts.merge(
+                        ref_line_flow_ts, on="time", how="outer"
+                    )
+                    info_col.markdown("### Observed vs Modeled Flow")
```
[the remark this answers is not in the document]
Comment (zherbz) [reply app/src/pages/single_event.py:603]: Yes, obs and mod flow are designed to be in the same plot. No data is shown for the obs gage hydrograph because nothing currently exists for the May 1990 event ID. https://ffrd.cloud.dewberryanalytics.com/projects/trinity/bucketview?bucket=trinity-pilot&key=stac%2Fprod-support%2Fpq-test%2Fgage%3D08044800%2F

File: app/src/pages/single_event.py [hunk 2]
```diff
@@ -401,39 +527,452 @@ def single_event():
...
+            num_models = 1
+            st.session_state["bc_lines_filtered"] = st.bc_lines[
+                st.bc_lines["model"] == st.session_state["model_id"]
+            ]
+            num_bc_lines = len(st.session_state["bc_lines_filtered"])
+
+    with col_about:
+        about_popover()
+    with col_dams:
+        map_popover(
```
[the remark this answers is not in the document]
Comment (zherbz) [reply app/src/pages/single_event.py:872]: This had to be moved to its current spot in order to filter the drop down widgets dynamically with map selections

File: app/src/pages/single_event.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zherbz) [reply app/src/pages/single_event.py:603]: The gage data needs to be migrated into the same structure as modeled data so that it can be queried in the same way. Currently the event_id differs from the modeled data and the observed data due differing ETL methods.

Example: Model = Bridgeport, Gage = 08042800, Event = Oct 2015
- Model Event ID = calibration_oct2015
- Gage Event ID = oct2015
- The model also has several more event IDs that the gage does not

[the remark this answers is not in the document]
Comment (zherbz) [reply app/src/pages/single_event.py:632]: Reference line plots are now separated into two plots of Obs vs Mod for both WSE and Flow. However, we currently have no data for Obs WSE so this will only plot the trace for Mod WSE.

**Comment by zherbz** [comment]
@thwllms ready for back check
